lib/go-tc/deliveryservices.go

```diff
@@ -846,7 +846,29 @@ func (ds *DeliveryServiceV4) RemoveLD1AndLD2() DeliveryServiceV4 {
 	return *ds
 }
 
-// DowngradeToV3 converts the 4.x DS to a 3.x DS.
+// DowngradeToV2 converts a 4.x DS to a 2.x DS.
+func (ds *DeliveryServiceV4) DowngradeToV2() DeliveryServiceNullable {
```

Comment:
You don't need to write this whole method, I think, the v2 DS can be accessed directly through `DeliveryServiceNullableV30.DeliveryServiceNullableV15`. That's what the `/deliveryservices` handler does.